src/core/namespace.c

```diff
@@ -526,15 +525,9 @@ int setup_namespace(
                 bool protect_sysctl,
                 bool protect_cgroups,
                 ProtectHome protect_home,
-                ProtectSystem protect_system,
-                unsigned long mount_flags) {
-
-        BindMount *m, *mounts = NULL;
+                ProtectSystem protect_system)
+{
```

Comment:
I can confirm: this commit fixes https://github.com/systemd/systemd/pull/4018#issuecomment-244285402
But I think we should squash https://github.com/systemd/systemd/pull/4185/commits/50d1975af392d2b999ced8444ace28c8fe0a46f7 and https://github.com/systemd/systemd/pull/4185/commits/1be5c8d176c94b460432d6218deb88e97bae9472